docs/src/simulation_tips.md

```diff
@@ -0,0 +1,153 @@
+
+```julia
+u, v, w = model.velocities
+u² = ComputedField(u^2)
+KE = ComputedField((u^2 + v^2 + w^2)/2)
+compute!(u²)
+compute!(KE)
+```
+
+There are two approaches to bypass this issue. The first is to nest `ComputedField`s. For example,
+we can make `KE` be computed on GPU by defining it as
+```julia
+u, v, w = model.velocities
+u² = ComputedField(u^2)
+v² = ComputedField(v^2)
+w² = ComputedField(w^2)
+uplusv = ComputedField(u² + v²)
+KE = ComputedField((uplusv + w²)/2)
+compute!(KE)
+```
+
+This is a simple workaround that is especially suited for the development stage of a simulation.
+However, when running this on a GPU, the code will iterate over the whole domain 5 times to
+calculate `KE` (one for each computed field defined), which is not very efficient.
+
+A different way to calculate `KE` is by using `KernelComputedField`s, where the
+user manually specifies the computing kernel to the compiler. The advantage of this method is that
+it's more efficient (the code will only iterate once over the domain in order to calculate `KE`),
+but the disadvantage is that this requires that the has some knowledge of Oceananigans operations
+and how they should be performed on a C-grid. For example calculating `KE` with this approach would
+look like this:
+
+```julia
+using Oceananigans.Operators
+using KernelAbstractions: @index, @kernel
+using Oceananigans.Grids: Center, Face
+using Oceananigans.Fields: KernelComputedField
+
+@inline ψ²(i, j, k, grid, ψ) = @inbounds ψ[i, j, k]^2
+
+@kernel function kinetic_energy_ccc!(tke, grid, u, v, w)
+    i, j, k = @index(Global, NTuple)
+    @inbounds tke[i, j, k] = (
+                              ℑxᶜᵃᵃ(i, j, k, grid, ψ², u) + # Interpolate x to center
+                              ℑyᵃᶜᵃ(i, j, k, grid, ψ², v) + # Interpolate y to center
+                              ℑzᵃᵃᶜ(i, j, k, grid, ψ², w)   # Interpolate z to center
+                             ) / 2
+end
+
+KE = KernelComputedField(Center, Center, Center, kinetic_energy_ccc!, model;
+                         computed_dependencies=(u, v, w))
+```
+
+It may be useful to know that there are some kernels already defined for commonly-used diagnostics
+in packages that are companions to Oceananigans. For example
+[Oceanostics.jl](https://github.com/tomchor/Oceanostics.jl/blob/13d2ba5c48d349c5fce292b86785ce600cc19a88/src/TurbulentKineticEnergyTerms.jl#L23-L30)
+and
+[LESbrary.jl](https://github.com/CliMA/LESbrary.jl/blob/master/src/TurbulenceStatistics/shear_production.jl).
+Users should first look there before writing any kernel by hand and are always welcome to [start an
+issue on Github](https://github.com/CliMA/Oceananigans.jl/issues/new) if they need help to write a
+different kernel.
+
+
+### Try to decrease the memory-use of your runs
+
+GPU runs are generally memory-limited, so it's good to both keep track of and try to reduce the size of your runs. Useful tips in this regard are
```

Comment:
There is an nvidia command that helps to profile.  Should we mention an example so the user has an idea where to start?